src/runtime-rs/crates/resource/src/network/network_pair.rs

```diff
@@ -177,3 +177,100 @@ pub async fn get_link_by_name(
 
     Ok(link::get_link_from_message(msg))
 }
+
+#[cfg(test)]
+mod tests {
+    use nix::unistd::getuid;
+    use rtnetlink::Handle;
+    use scopeguard::defer;
+
+    use crate::network::network_model::{tc_filter_model::fetch_index, TC_FILTER_NET_MODEL_STR};
+
+    use super::*;
+
+    async fn delete_link(handle: &Handle, name: &str) -> Result<(), rtnetlink::Error> {
+        // delete the manually created links
+        let link_index = fetch_index(&handle, name)
+            .await
+            .expect("failed to fetch index");
+        handle.link().del(link_index).execute().await
+    }
+
+    // this ut tests create_link() and get_link_by_name()
+    #[actix_rt::test]
+    async fn test_utils() {
+        if let Ok((conn, handle, _)) =
+            rtnetlink::new_connection().context("failed to create netlink connection")
+        {
+            let thread_handler = tokio::spawn(conn);
+            defer!({
+                thread_handler.abort();
+            });
+
+            // there is no links created, so err
+            assert!(get_link_by_name(&handle, "kata_test_1").await.is_err());
+            assert!(get_link_by_name(&handle, "kata_test_2").await.is_err());
+            assert!(get_link_by_name(&handle, "kata_test_3").await.is_err());
+
+            // create link needs to be root
+            if !getuid().is_root() {
+                return;
+            }
+
+            assert!(create_link(&handle, "kata_test_1", 2).await.is_ok());
+            assert!(create_link(&handle, "kata_test_2", 3).await.is_ok());
+            assert!(create_link(&handle, "kata_test_3", 4).await.is_ok());
+
+            assert!(get_link_by_name(&handle, "kata_test_1").await.is_ok());
+            assert!(get_link_by_name(&handle, "kata_test_2").await.is_ok());
+            assert!(get_link_by_name(&handle, "kata_test_3").await.is_ok());
+
+            assert!(delete_link(&handle, "kata_test_1").await.is_ok());
+            assert!(delete_link(&handle, "kata_test_2").await.is_ok());
+            assert!(delete_link(&handle, "kata_test_3").await.is_ok());
```

Comment:
I think it's better to move line210-213 after line 230.

Reply:
Fixed

Reply:
Fixed